packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -34,35 +35,77 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         }
 
         const { uid } = sessionToken;
-        const { recoveryKeyId, recoveryData } = request.payload;
+        const { recoveryKeyId, recoveryData, enabled } = request.payload;
 
-        await db.createRecoveryKey(uid, recoveryKeyId, recoveryData);
+        await db.createRecoveryKey(uid, recoveryKeyId, recoveryData, enabled);
 
         log.info('account.recoveryKey.created', { uid });
 
-        await request.emitMetricsEvent('recoveryKey.created', { uid });
+        if (enabled) {
+          await request.emitMetricsEvent('recoveryKey.created', { uid });
+
+          const account = await db.account(uid);
+
+          const { acceptLanguage, clientAddress: ip, geo, ua } = request.app;
+          const emailOptions = {
+            acceptLanguage,
+            ip,
+            location: geo.location,
+            timeZone: geo.timeZone,
+            uaBrowser: ua.browser,
+            uaBrowserVersion: ua.browserVersion,
+            uaOS: ua.os,
+            uaOSVersion: ua.osVersion,
+            uaDeviceType: ua.deviceType,
+            uid,
+          };
+
+          await mailer.sendPostAddAccountRecoveryEmail(
+            account.emails,
+            account,
+            emailOptions
+          );
+        }
 
-        const account = await db.account(uid);
+        return {};
+      },
+    },
+    {
+      method: 'POST',
+      path: '/recoveryKey/verify',
+      options: {
+        auth: {
+          strategy: 'sessionToken',
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: validators.recoveryKeyId,
+          },
+        },
+      },
+      handler: async function(request) {
+        log.begin('verifyRecoveryKey', request);
 
-        const { acceptLanguage, clientAddress: ip, geo, ua } = request.app;
-        const emailOptions = {
-          acceptLanguage,
-          ip,
-          location: geo.location,
-          timeZone: geo.timeZone,
-          uaBrowser: ua.browser,
-          uaBrowserVersion: ua.browserVersion,
-          uaOS: ua.os,
-          uaOSVersion: ua.osVersion,
-          uaDeviceType: ua.deviceType,
-          uid,
-        };
+        const sessionToken = request.auth.credentials;
+        const { uid } = sessionToken;
 
-        await mailer.sendPostAddAccountRecoveryEmail(
-          account.emails,
-          account,
-          emailOptions
-        );
+        if (sessionToken.tokenVerificationId) {
+          throw errors.unverifiedSession();
+        }
+
+        // This route can let you check if a key is valid therefore we
+        // rate limit it.
+        await customs.checkAuthenticated(request, uid, 'getRecoveryKey');
```

Comment:
I'm not familiar with default rate limit rules, does `checkAuthenticated` come with that?